modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/HeavyOrLightWeightAppControlTestCase.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.runtime.module.deployment.internal;
+
+import com.github.valfirst.slf4jtest.LoggingEvent;
+import com.github.valfirst.slf4jtest.TestLogger;
+import io.qameta.allure.Description;
+import io.qameta.allure.Feature;
+import io.qameta.allure.Story;
+import org.junit.Test;
+import org.junit.runners.Parameterized;
+import org.mule.maven.client.api.model.MavenConfiguration.MavenConfigurationBuilder;
+import org.mule.runtime.module.deployment.impl.internal.builder.ApplicationFileBuilder;
+import org.mule.runtime.module.deployment.impl.internal.builder.ArtifactPluginFileBuilder;
+
+import java.io.File;
+import java.net.URL;
+import java.nio.file.Paths;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+import static com.github.valfirst.slf4jtest.TestLoggerFactory.getTestLogger;
+import static java.util.Arrays.asList;
+import static java.util.stream.Collectors.toList;
+import static org.apache.commons.io.FileUtils.iterateFiles;
+import static org.apache.commons.io.filefilter.HiddenFileFilter.VISIBLE;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.endsWith;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.startsWith;
+import static org.junit.Assert.assertThat;
+import static org.mule.maven.client.api.model.MavenConfiguration.newMavenConfigurationBuilder;
+import static org.mule.maven.client.api.model.RemoteRepository.newRemoteRepositoryBuilder;
+import static org.mule.runtime.container.api.MuleFoldersUtil.getAppFolder;
+import static org.mule.runtime.container.api.MuleFoldersUtil.getMuleBaseFolder;
+import static org.mule.runtime.container.internal.ClasspathModuleDiscoverer.EXPORTED_CLASS_PACKAGES_PROPERTY;
+import static org.mule.runtime.globalconfig.api.GlobalConfigLoader.setMavenConfig;
+import static org.mule.test.allure.AllureConstants.DeploymentTypeFeature.DEPLOYMENT_TYPE;
+import static org.mule.test.allure.AllureConstants.DeploymentTypeFeature.DeploymentTypeStory.HEAVYWEIGHT;
+import static org.mule.test.allure.AllureConstants.DeploymentTypeFeature.DeploymentTypeStory.LIGHTWEIGHT;
+
+@Feature(DEPLOYMENT_TYPE)
+public class HeavyOrLightWeightAppControlTestCase extends AbstractApplicationDeploymentTestCase {
```

Comment:
Add different classes for each case with different stories in @story. Possibly create an abstract class for common code. Or used a parametized test. Notice that the three tests are basically the same. They have a lot of code in common. Try to add parameters for those tests.